### PR TITLE
Update GitLab provider docs

### PR DESCRIPTION
**Pull request by bradjones1**
## Summary

Updates GitLab provider docs to reference self-hosted GitLab and provide additional clarity on required scopes.

**Checklist**:
- [ ] add related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

**Review by desimone — COMMENTED** [review]
Thank you for the PR @bradjones1 ! 

It has been a while since I have personally tested Gitlab, but I seem to remember that one set of scopes was available for on-prem (`read_api`) but was not available for the hosted version (and therefore required full `api`).

The directory synchronization functionality (e.g. fetching groups) should work for both modes with the idp provider set as gitlab. Are you not finding that to be the case?

**Comment by desimone** [comment]
@bradjones1 ping :)

**Comment by bradjones1** [comment]
I'm off this week but I will loop back around on this.

**Comment by travisgroth** [comment]
Closing this for now.  Feel free to re-open when ready.

**Comment by bradjones1** [comment]
Please re-open :-)

File: docs/docs/identity-providers/gitlab.md [hunk 1]
```diff
@@ -25,9 +25,7 @@ Field        | Description
 ------------ | --------------------------------------------
 Name         | The name of your web app
 Redirect URI | `https://${authenticate_service_url}/oauth2/callback`
-Scopes       | **Must** select the same as **[identity scopes]** option
-
-If no scopes are set, you **must** select **openid**, **api**, **profile** and **email**.
+Scopes       | `openid` required; `read_api`, `profile`, `email` as necessary for your policies.
```
[the remark this answers is not in the document]
Comment (bradjones1) [reply docs/docs/identity-providers/gitlab.md:28]: I just verified on latest self-hosted that `read_api` is accessible.

File: docs/docs/identity-providers/gitlab.md [hunk 2]
```diff
@@ -52,20 +50,35 @@ The format of the `idp_service_account` for Gitlab is a base64-encoded JSON docu
...
+Self-hosted CE/EE instances should be configured as a generic OpenID Connect provider:
 
-Please be aware that [Group ID](https://docs.gitlab.com/ee/api/groups.html#details-of-a-group) will be used to affirm group(s) a user belongs to.
-
-[identity scopes]: ../../reference/readme.md#identity-provider-scopes
+```
+idp_provider: oidc
+idp_client_id: "REACTED"
+idp_client_secret: "REDACTED"
+idp_scopes: openid,email // Intersects with scopes
```
[the remark this answers is not in the document]
Comment (bradjones1) [reply docs/docs/identity-providers/gitlab.md:81]: No; AFAIK `offline_access` is not an RFC defined spec but more [convention](https://stackoverflow.com/questions/30637984/what-does-offline-access-in-oauth-mean). For what it's worth, I don't think refresh tokens are even pertinent here (in a default configuration, definitely not on gitlab.com) in so far as GitLab auth tokens are [valid indefinitely and no refresh tokens are issued](https://gitlab.com/gitlab-org/gitlab/-/issues/21745#note_215137958).

**Comment by bradjones1** [comment]
@desimone Rebased this and addressed your comments, thanks.

File: docs/docs/identity-providers/gitlab.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (desimone) [reply docs/docs/identity-providers/gitlab.md:81]: Thank you for the clarification

**Review by desimone — APPROVED** [review]
Thank you for the contribution @bradjones1 ; apologies, I didn't see your update till just now.

**Pull request merged**